src/encord_active/app/actions_page/export_filter.py

```diff
@@ -580,32 +582,52 @@ def render_relabel_button(
         disabled=is_disabled,
+        on_click=lambda: current_form.set(CurrentForm.RELABEL),
     )
-    if relabel_data:
-        label = st.empty()
-        label.text("Sending the selected data to labeling.")
-        update_progress_bar, clear_progress_bar = render_progress_bar()
-        update_progress_bar(0)  # Show progress bar
 
-        # Get the label rows to send to relabel
-        unique_lr_hashes = list(set(str(_id).split("_", maxsplit=1)[0] for _id in filtered_df["identifier"].to_list()))
+    if current_form.value != CurrentForm.RELABEL:
+        return
 
-        encord_project = get_encord_project(project_meta["ssh_key_path"], project_meta["project_hash"])
-        for index, label_row in enumerate(encord_project.list_label_rows_v2(label_hashes=unique_lr_hashes), start=1):
-            label_row.workflow_reopen()
-            update_progress_bar(index / len(unique_lr_hashes))
+    # Get the label rows to submit for labeling
+    all_identifiers = filtered_df["identifier"].to_list()
+    unique_lr_hashes = list(set(str(_id).split("_", maxsplit=1)[0] for _id in all_identifiers))
 
-        clear_progress_bar()
-        label.empty()
+    form_placeholder = st.empty()  # Show the confirmation form
+    with form_placeholder.form("relabel_confirmation_form"):
+        label = st.empty()
         label.text(
-            f"Data successfully sent to labeling in Encord Annotate.\n"
-            f"A total of {len(unique_lr_hashes)} tasks were assigned to the first labeling stage."
+            # Next line should stay commented while 'all_identifiers' has a wrong amount of items
+            # f"You have selected {len(all_identifiers)} items to submit for labeling in Encord Annotate.\n"
+            f"A total of {len(unique_lr_hashes)} tasks are going to be assigned to the first labeling stage.\n"
+            f"Are you sure you want to proceed?"
         )
 
+        if not st.form_submit_button("Confirm"):
+            return None
+    form_placeholder.empty()  # Disappear the form after clicking the confirmation button
+
+    # Start the submission of the selected label rows for labeling
+    label = st.empty()
+    label.text("Sending the selected data to labeling.")
+    update_progress_bar, clear_progress_bar = render_progress_bar()
+    update_progress_bar(0)  # Show progress bar
+
+    encord_project = get_encord_project(project_meta["ssh_key_path"], project_meta["project_hash"])
+    for index, label_row in enumerate(encord_project.list_label_rows_v2(label_hashes=unique_lr_hashes), start=1):
+        label_row.workflow_reopen()
+        update_progress_bar(index / len(unique_lr_hashes))
+
+    clear_progress_bar()
+    label.empty()
+    label.text(
+        f"Data successfully submitted for labeling in Encord Annotate.\n"
+        f"A total of {len(unique_lr_hashes)} tasks have been assigned to the first labeling stage."
+    )
```

Comment:
Think that we should add a link to the project here as well for people to jump directly to the project.

Reply:
great idea!